pkg/flowcli/services/scripts.go

```diff
@@ -49,25 +52,49 @@ func NewScripts(
 }
 
 // Execute executes a Cadence script from a file.
-func (s *Scripts) Execute(scriptFilename string, args []string, argsJSON string) (cadence.Value, error) {
-	script, err := util.LoadFile(scriptFilename)
+func (s *Scripts) Execute(scriptPath string, args []string, argsJSON string, network string) (cadence.Value, error) {
+	script, err := util.LoadFile(scriptPath)
 	if err != nil {
 		return nil, err
 	}
 
-	return s.execute(script, args, argsJSON)
+	return s.execute(script, args, argsJSON, scriptPath, network)
 }
 
 // Execute executes a Cadence script from a source code string.
 func (s *Scripts) ExecuteWithCode(code []byte, args []string, argsJSON string) (cadence.Value, error) {
-	return s.execute(code, args, argsJSON)
+	return s.execute(code, args, argsJSON, "", "")
 }
 
-func (s *Scripts) execute(code []byte, args []string, argsJSON string) (cadence.Value, error) {
+func (s *Scripts) execute(code []byte, args []string, argsJSON string, scriptPath string, network string) (cadence.Value, error) {
 	scriptArgs, err := flowcli.ParseArguments(args, argsJSON)
 	if err != nil {
 		return nil, err
 	}
 
+	resolver, err := contracts.NewResolver(code)
+	if err != nil {
+		return nil, err
+	}
+
+	if resolver.ImportExists() {
+		if s.project == nil {
+			return nil, fmt.Errorf("missing configuration, initialize it: flow init")
+		} else if network == "" {
+			return nil, fmt.Errorf("missing network, specify which network to use to resolve imports in script code")
+		} else if scriptPath == "" { // when used as lib with code we don't support imports
+			return nil, fmt.Errorf("resolving imports in scripts not supported")
+		}
```

Comment:
I would just make each a separate `if` statement, no need to have an `else` chain if each block has a `return` statement. Feels a bit easier to read IMO

Reply:
I was trying to save up lines but you are right I will change